diff --git a/node_modules/react-native-share-menu/android/build.gradle b/node_modules/react-native-share-menu/android/build.gradle
index 1234567..abcdefg 100644
--- a/node_modules/react-native-share-menu/android/build.gradle
+++ b/node_modules/react-native-share-menu/android/build.gradle
@@ -1,13 +1,13 @@
 apply plugin: 'com.android.library'

 android {
-    compileSdkVersion 29
-    buildToolsVersion "29.0.2"
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion

     defaultConfig {
-        minSdkVersion 16
-        targetSdkVersion 29
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.targetSdkVersion
         versionCode 1
         versionName "1.0"
         ndk {
diff --git a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
index 09abd7b..887b4e4 100644
--- a/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
+++ b/node_modules/react-native-share-menu/android/src/main/java/com/meedan/ShareMenuModule.java
@@ -58,7 +58,18 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
     WritableMap data = Arguments.createMap();
     data.putString(MIME_TYPE_KEY, type);
 
-    if (Intent.ACTION_SEND.equals(action)) {
+    if (Intent.ACTION_PROCESS_TEXT.equals(action)) {
+      if ("text/plain".equals(type)) {
+        data.putString(DATA_KEY, intent.getStringExtra(Intent.EXTRA_PROCESS_TEXT));
+        return data;
+      }
+
+      Uri fileUri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+      if (fileUri != null) {
+        data.putString(DATA_KEY, fileUri.toString());
+        return data;
+      }
+    } else if (Intent.ACTION_SEND.equals(action)) {
       if ("text/plain".equals(type)) {
         data.putString(DATA_KEY, intent.getStringExtra(Intent.EXTRA_TEXT));
         return data;
@@ -92,19 +103,6 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
       return;
     }
 
-    // If this isn't the root activity then make sure it is
-    if (!currentActivity.isTaskRoot()) {
-      Intent newIntent = new Intent(currentActivity.getIntent());
-      newIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-      currentActivity.startActivity(newIntent);
-
-      ReadableMap shared = extractShared(newIntent);
-      successCallback.invoke(shared);
-      clearSharedText();
-      currentActivity.finish();
-      return;
-    }
-
     Intent intent = currentActivity.getIntent();
     
     ReadableMap shared = extractShared(intent);
@@ -136,6 +134,7 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
 
     if ("text/plain".equals(type)) {
       intent.removeExtra(Intent.EXTRA_TEXT);
+      intent.removeExtra(Intent.EXTRA_PROCESS_TEXT);
       return;
     }
 
@@ -163,4 +162,14 @@ public class ShareMenuModule extends ReactContextBaseJavaModule implements Activ
     // Update intent in case the user calls `getSharedText` again
     currentActivity.setIntent(intent);
   }
+
+  @ReactMethod
+  public void addListener(String eventName) {
+    // Keep: Required for RN built in Event Emitter Calls.
+  }
+
+  @ReactMethod
+  public void removeListeners(Integer count) {
+    // Keep: Required for RN built in Event Emitter Calls.
+  }
 }
diff --git a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
index f42bce6..5812889 100644
--- a/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
+++ b/node_modules/react-native-share-menu/ios/ReactShareViewController.swift
@@ -12,8 +12,8 @@ import RNShareMenu
 class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactShareViewDelegate {
   func sourceURL(for bridge: RCTBridge!) -> URL! {
 #if DEBUG
-    return RCTBundleURLProvider.sharedSettings()?
-      .jsBundleURL(forBundleRoot: "index.share", fallbackResource: nil)
+    return RCTBundleURLProvider.sharedSettings()
+      .jsBundleURL(forBundleRoot: "index")
 #else
     return Bundle.main.url(forResource: "main", withExtension: "jsbundle")
 #endif
@@ -25,24 +25,11 @@ class ReactShareViewController: ShareViewController, RCTBridgeDelegate, ReactSha
     let bridge: RCTBridge! = RCTBridge(delegate: self, launchOptions: nil)
     let rootView = RCTRootView(
       bridge: bridge,
-      moduleName: "ShareMenuModuleComponent",
+      moduleName: "VocablyProShare",
       initialProperties: nil
     )
 
-    rootView.backgroundColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
-    backgroundColorSetup: if let backgroundColorConfig = Bundle.main.infoDictionary?[REACT_SHARE_VIEW_BACKGROUND_COLOR_KEY] as? [String:Any] {
-      if let transparent = backgroundColorConfig[COLOR_TRANSPARENT_KEY] as? Bool, transparent {
-        rootView.backgroundColor = nil
-        break backgroundColorSetup
-      }
-
-      let red = backgroundColorConfig[COLOR_RED_KEY] as? Float ?? 1
-      let green = backgroundColorConfig[COLOR_GREEN_KEY] as? Float ?? 1
-      let blue = backgroundColorConfig[COLOR_BLUE_KEY] as? Float ?? 1
-      let alpha = backgroundColorConfig[COLOR_ALPHA_KEY] as? Float ?? 1
-
-      rootView.backgroundColor = UIColor(red: CGFloat(red), green: CGFloat(green), blue: CGFloat(blue), alpha: CGFloat(alpha))
-    }
+    rootView.backgroundColor = UIColor.systemBackground
 
     self.view = rootView
 
